### PR TITLE
 Secure HTTP Strict Transport Security

### PR DESCRIPTION
**Pull request by Zakaria-Latif**
Secure HTTP Strict Transport Security KB

### CODE REVIEW

File: WEB_SERVICE/WEB/_SECURE/SECURE_HTTP_HSTS/description.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+**HTTP Strict Transport Security (HSTS) Overview**
+
+HTTP Strict Transport Security (HSTS) is a web security policy mechanism that ensures users interact with the web server only through secure (HTTPS) connections. This is achieved by the server communicating the HSTS policy to user agents (such as web browsers) via the "Strict-Transport-Security" HTTP response header. The HSTS policy specifies a duration during which the user agent is required to access the server securely.
+
+**Secure HSTS Implementation**
+
+The application has implemented a secure HSTS policy to enforce the use of HTTPS and protect against man-in-the-middle attacks. This configuration ensures that all communications between the client and the server are encrypted, enhancing overall security.
```
Comment (PiranhaSa) [inline WEB_SERVICE/WEB/_SECURE/SECURE_HTTP_HSTS/description.md:7]: Don't you find these two repetitive ?

File: WEB_SERVICE/WEB/_SECURE/SECURE_HTTP_HSTS/description.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+1. **Enforced HTTPS**: The HSTS policy automatically converts any insecure HTTP links to secure HTTPS links, ensuring all interactions are encrypted.
+   
+2. **Prevention of Insecure Access**: If a secure connection cannot be established (e.g., due to an invalid TLS certificate), the user agent will block access, preventing potential security risks.
+
+3. **Configurable Duration**: The HSTS policy specifies a long enough duration to ensure that users will continue to access the application securely over time.
+
+4. **Subdomain Support**: If applicable, the HSTS policy includes the "includeSubDomains" directive, extending security to all subdomains.
+
+5. **Ongoing Monitoring**: Continuous monitoring is in place to ensure that the HSTS settings are correctly enforced and that the security of TLS certificates is maintained.
```
Comment (nmasdoufi-ol) [inline WEB_SERVICE/WEB/_SECURE/SECURE_HTTP_HSTS/description.md:19]:
```suggestion
5. **Ongoing Monitoring**: Continuous monitoring is in place to ensure that the HSTS settings are correctly enforced and that the security of TLS certificates is maintained.

6. **Preload Option**: The application can be submitted to the HSTS preload list, which allows browsers to know in advance that the site should only be accessed over HTTPS, even on the first visit.
```

Comment (PiranhaSa) [inline WEB_SERVICE/WEB/_SECURE/SECURE_HTTP_HSTS/description.md:19]:
```suggestion
5. **Ongoing Monitoring**: Continuous monitoring is in place to ensure that the HSTS settings are correctly enforced and that the security of TLS certificates is maintained.
```
I don't think this is a feature of HSTS

**Review by ybadaoui-ostorlab — APPROVED** [review]

File: WEB_SERVICE/WEB/_SECURE/SECURE_HTTP_HSTS/description.md [hunk 3]
```diff
@@ -0,0 +1,29 @@
+**HTTP Strict Transport Security (HSTS) Overview**
+
```
Comment (3asm) [inline WEB_SERVICE/WEB/_SECURE/SECURE_HTTP_HSTS/description.md:2]:
```suggestion
```
No need to have a title here.

**Review by 3asm — CHANGES_REQUESTED** [review]

**Review by 3asm — APPROVED** [review]

**Pull request merged**